### ISSUE TITLE
GH Actions: timeout during Package Control installation

### ISSUE THREAD

**Issue by giampaolo** [SublimeText/UnitTesting#208]
Hello!
I am trying to setup GH Actions and unfortunately I get this error:

```
....
2022-12-15T23:36:46.3254891Z copy the package to sublime package directory
2022-12-15T23:36:46.3571658Z /home/runner/work/sysconf/sysconf
2022-12-15T23:36:46.6617191Z Installing UnitTesting from https://github.com/SublimeText/UnitTesting@1.5.9 to /home/runner/.config/sublime-text/Packages/UnitTesting
2022-12-15T23:36:47.5030406Z Installing coverage from https://github.com/codexns/sublime-coverage@1.0.0 to /home/runner/.config/sublime-text/Packages/coverage
2022-12-15T23:36:48.1481026Z ##[group]Run bash $GITHUB_ACTION_PATH/../../scripts/install_package_control.sh
2022-12-15T23:36:48.1481497Z [36;1mbash $GITHUB_ACTION_PATH/../../scripts/install_package_control.sh[0m
2022-12-15T23:36:48.1572217Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2022-12-15T23:36:48.1572529Z env:
2022-12-15T23:36:48.1572770Z   SUBLIME_TEXT_VERSION: 4
2022-12-15T23:36:48.1573040Z   SUBLIME_TEXT_ARCH: x64
2022-12-15T23:36:48.1573373Z   SUBLIME_TEXT_PACKAGES: /home/runner/.config/sublime-text/Packages
2022-12-15T23:36:48.1573703Z   DISPLAY: :1
2022-12-15T23:36:48.1573945Z   PACKAGE: sysconf
2022-12-15T23:36:48.1574182Z ##[endgroup]
2022-12-15T23:36:48.5935756Z creating Package Control.sublime-settings
2022-12-15T23:40:00.8194433Z ....................................Timeout: Fail to install Package Control.
2022-12-15T23:40:00.8248200Z ##[error]Process completed with exit code 1.
```

My config looks like this:

```yml
name: sublime-tests
on: [push, pull_request]
jobs:
  run-tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: SublimeText/UnitTesting/actions/setup@v1
      - uses: SublimeText/UnitTesting/actions/run-tests@v1

  run-syntax-tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: SublimeText/UnitTesting/actions/setup@v1
      - uses: SublimeText/UnitTesting/actions/run-syntax-tests@v1
```

I'm really not sure what to make of this. I'm not sure if it's related, but my Github repo is private.

Thanks!

**Comment by randy3k** [comment]
Did you try rerunning the tests? Random failure may sometime happen.

**Comment by giampaolo** [comment]
Yeah, nothing changes. :(
This (new) error is from the `run-syntax-tests:` job:

```
2022-12-16T11:46:05.2893103Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2022-12-16T11:46:05.2893349Z env:
2022-12-16T11:46:05.2893528Z   SUBLIME_TEXT_VERSION: 4
2022-12-16T11:46:05.2893739Z   SUBLIME_TEXT_ARCH: x64
2022-12-16T11:46:05.2894110Z   SUBLIME_TEXT_PACKAGES: /home/runner/.config/sublime-text/Packages
2022-12-16T11:46:05.2894363Z   DISPLAY: :1
2022-12-16T11:46:05.2894550Z   PACKAGE: sysconf
2022-12-16T11:46:05.2894735Z ##[endgroup]
2022-12-16T11:46:37.3664621Z Wait for tests output..................................
2022-12-16T11:47:09.4035281Z Wait for tests output..................................
2022-12-16T11:47:39.4427242Z Wait for tests output..................................
2022-12-16T11:47:39.4429713Z Timeout: Could not obtain tests output.
2022-12-16T11:47:39.4430226Z Maybe Sublime Text is not responding or the tests output is being written to the wrong file.
2022-12-16T11:47:39.4442445Z ##[error]Process completed with exit code 1.
2022-12-16T11:47:39.4495895Z Post job cleanup.
2022-12-16T11:47:39.6542324Z [command]/usr/bin/git version
```

**Comment by giampaolo** [comment]
Again, I suspect this may have to do with the fact that the repo is private. 
I'm taking a look at https://github.com/SublimeText/UnitTesting/tree/master/actions/, but I'm not familiar with this sort of thing. Perhaps one those scripts assume the repo is public, tries to download it and fail?
Has UnitTesting ever been used or tested with private repos?

**Comment by giampaolo** [comment]
Quote:

```
2022-12-16T11:46:37.3664621Z Wait for tests output..................................
2022-12-16T11:47:09.4035281Z Wait for tests output..................................
2022-12-16T11:47:39.4427242Z Wait for tests output..................................
2022-12-16T11:47:39.4429713Z Timeout: Could not obtain tests output.
```

Another possible reason which came to mind is this: my ST plugins are **not** defined in the root directory:

```bash
$ lstree home/.config/sublime-text/Packages/User/
plugin_1.py
plugin_2.py
unittesting.json
tests/test_plugin_1.py
tests/test_plugin_2.py
$
$ cat home/.config/sublime-text/Packages/User/unittesting.json
{
    "tests_dir" : "tests",
    "pattern" : "test*.py",
    "verbosity": 10,
    "capture_console": true,
    "reload_package_on_testing": false
}
```

I think I need to tell the GH actions config file where the tests are and/or how to run them.

**Comment by randy3k** [comment]
In default, the package is assumed to be placed the root directory of the repo, or you'd need to specify `package-path` https://github.com/SublimeText/UnitTesting/blob/master/actions/setup/action.yaml#L19

**Comment by randy3k** [comment]
Additionally, it may be better to name your package something other than `User`.

**Comment by giampaolo** [comment]
Thank you for your help Randy. I have added this:

```yml
      - uses: actions/checkout@v3
      - uses: SublimeText/UnitTesting/actions/setup@v1
        with:
          package-name: User
          package-path: ./home/.config/sublime-text/Packages/User
```

That didn't solve the problem but at least the error changed (that's a progress :)):

```
...
2022-12-16T18:53:51.2439963Z env:
2022-12-16T18:53:51.2440153Z   SUBLIME_TEXT_VERSION: 4
2022-12-16T18:53:51.2440385Z   SUBLIME_TEXT_ARCH: x64
2022-12-16T18:53:51.2440677Z   SUBLIME_TEXT_PACKAGES: /home/runner/.config/sublime-text/Packages
2022-12-16T18:53:51.2440938Z   DISPLAY: :1
2022-12-16T18:53:51.2441138Z   PACKAGE: User
2022-12-16T18:53:51.2441337Z ##[endgroup]
2022-12-16T18:53:55.4886625Z Wait for tests output.......
2022-12-16T18:53:55.4887289Z Start to read output...
2022-12-16T18:53:55.4888063Z No tests are found.
```

> Additionally, it may be better to place your package in somewhere else other than User.

You're right, but this is not a real ST package. It's a (sort of big) collections of personal plugins / configs which I keep under revision control, so that my ST setup is fully portable and replicable everywhere. To give you an idea, it looks like this:

```bash
~/svn/sysconf $ git remote 
origin
~/svn/sysconf $ ls home/.config/sublime-text/Packages/User/
ANSIescape                         git_grep.py                                         pypaths.py
Log Highlight                      git_log.py                                          Python.sublime-settings
Projects                           git_show.py                                         quick_find.py
snippets                           git_squash.py                                       refresh.py
tests                              git_stash.py                                        reindent.py
UnitTesting                        git_status.py                                       rm_comments.py
__init__.py                        GitGutter.sublime-settings                          save_cursor_positions.py
Adaptive.sublime-theme             github.py                                           scroll.py
C.sublime-settings                 goto.py                                             selection.py
clear_console.py                   increase_decrease.py                                sidebar.py
Color Highlight.sublime-settings   INI.sublime-settings                                SingleTrailingNewLine.sublime-settings
create_snippet_from_sel.py         linters.py                                          smart_goto.py
Default (Linux).sublime-keymap     list_all_cmds.py                                    smart_paste.py
Default.sublime-commands           LSP-bash.sublime-settings                           smart_wrap_text.py
Default.sublime-theme              LSP-pyright.sublime-settings                        status_bar.py
Default Dark.sublime-theme         LSP.sublime-settings                                SublimeLint.sublime-settings
Distraction Free.sublime-settings  Monokai.sublime-color-scheme                        SublimeLinter.sublime-settings
edit_settings.py                   Monokai.tmTheme                                     tabs.py
events.py                          Monokai Pro (Filter Spectrum).sublime-color-scheme  Terminus.sublime-settings
exec_last_cmd.py                   move_focus.py                                       toggle.py
file.py                            Origami.sublime-settings                            TOML.sublime-settings
FileHistory.sublime-settings       OverrideAudit.status                                unittesting.json
find_results.py                    Package Control.sublime-settings                    utils.py
Flake8Lint.sublime-settings        Package Control.user-ca-bundle                      utils_base.py
Git.sublime-settings               PanePane.sublime-settings                           utils_command.py
git_branches.py                    Preferences.sublime-settings                        utils_git.py
git_checkout.py                    prettify.py                                         utils_input.py
git_commit.py                      pydoc.py                                            utils_settings.py
git_diff.py                        Pylinter.sublime-settings                           utils_ui.py
git_files.py                       pylinters.py                                        utils_view.py
```

After I install ST on a brand new system (and here is the kicker) I simply symlink `~/home/.config/sublime-text/Packages/User/` to that directory.  This has worked flawlessly for me so far, but this is why this is not a conventional ST package.

**Comment by giampaolo** [comment]
If this can help, here is how I'm running tests via the build system. My project config file has this:

```json
{
    // ...
    "build_systems": [
        {
            "name": "Sublime tests:",
            "target": "unit_testing",
            "package": "User",
            "variants": [
               {"name": "test_utils_base.py",  "pattern": "test_utils_base.py"},
               {"name": "test_utils_commands.py",  "pattern": "test_utils_commands.py"},
               {"name": "test_utils_settings.py",  "pattern": "test_utils_settings.py"},
               {"name": "test_utils_input.py",  "pattern": "test_utils_input.py"},
               {"name": "test_create_snippet_from_sel.py",  "pattern": "test_create_snippet_from_sel.py"},
           ]
        },
    ]
}
```

**Comment by deathaxe** [comment]
Fixed by 1.6.0
